### PR TITLE
Emit an event when there are clang-format warnings, so the user can act

### PR DESCRIPTION
**Pull request by alexeagle**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,10 +11,25 @@ Sample gulpfile.js:
...
+### Promoting warnings to errors
+If you want to enforce the formatting, so that other team members don't introduce
+code that gives you a warning, you can turn them into build errors by acting on
+the 'warning' event. For example, this task exits the build immediately:
+```js
+var format = require('gulp-clang-format');
+gulp.task('check-format', function() {
+  return gulp.src('*.js')
+	.pipe(format.checkFormat('file'))
+    .on('warning', function(e) { process.stdout.write(e.message); process.exit(1) });
```
Comment (mprobst) [inline README.md:28]: Would it be more convenient for users to call `.pipe(format.checkFormat('file', {failOnFormatError: true})`, and then have the pipeline emit an error? Or is that not possible thanks to Gulp?

**Comment by mprobst** [comment]
LGTMs otherwise, in case the `failOnFormatError` thing isn't possible.

**Comment by alexeagle** [comment]
It is possible to emit error instead, and add an optional parameter.
However, due to messy error handling, if you don't handle the error event it just breaks the pipeline. You need to have an on('error', {}) to properly deal with it - so it's longer than what I have here.
I also like the idea that user could do something else in the warning case, like run the clang-format command line themselves. So it's nice that we just emit 'warning' and it's up to them whether to treat as a warning or treat as an error.
Maybe it should be a gulp standard event?

**Comment by mprobst** [comment]
I see, that makes sense.

I don't know what a gulp standard event would be. I think I've seen 'warn' somewhere, but have no idea if that's more standard. Overall gulp error handling seems like a lost cause, let's cut our investment and move to broccoli ASAP.

**Comment by alexeagle** [comment]
thanks for review, merging as-is.

**Comment by alexeagle** [comment]
merged.

**Pull request closed without merging**